services/sonar/sonar-generic.service.js

```diff
@@ -0,0 +1,152 @@
+'use strict'
+
+const { metric } = require('../text-formatters')
+const SonarBase = require('./sonar-base')
+const { patternBase, queryParamSchema, getLabel } = require('./sonar-helpers')
+
+// This service is intended to be a temporary solution to avoid breaking
+// any existing users/badges that were utilizing the "other" Sonar metrics
+// with the Legacy Shields service implementation for Sonar badges.
+// The legacy implementation simply rendered a brightgreen badge with the value,
+// regardless of what the value actually was.
+//
+// See https://github.com/badges/shields/issues/3236 for more information.
+//
+// This service should gradually be replaced by services that handle
+// their respective metrics by providing badges with more context
+// (i.e. a red/error badge when there are multiple security issues)
+//
+// https://docs.sonarqube.org/latest/user-guide/metric-definitions
```

Comment:
I hope this provides sufficient info behind why this service exists. Definitely open to any suggestions here though, as I'm likely too close to this